Meadowlark-js/backends/meadowlark-postgresql-backend/src/repository/Delete.ts

```diff
@@ -32,41 +41,45 @@ export async function deleteDocumentById(
   { id, validate, traceId }: DeleteRequest,
   client: PoolClient,
 ): Promise<DeleteResult> {
-  const deleteResult: DeleteResult = { response: 'UNKNOWN_FAILURE' };
+  let deleteResult: DeleteResult = { response: 'UNKNOWN_FAILURE' };
 
   try {
     client.query('BEGIN');
 
-    // TODO - Reference validation to be added with RND-243
     if (validate) {
-      //         // Check for any references to the document to be deleted
-      //         const anyReferences: WithId<MeadowlarkDocument> | null = await mongoCollection.findOne(
-      //           onlyDocumentsReferencing(id),
-      //           onlyReturnId(session),
-      //         );
-      //         // Abort on validation failure
-      //         if (anyReferences) {
-      //           Logger.debug(
-      //             `mongodb.repository.Delete.deleteDocumentById: Deleting document id ${id} failed due to existing references`,
-      //             traceId,
-      //           );
-      //           // Get the DocumentIdentities of up to five referring documents for failure message purposes
-      //           const referringDocuments = await mongoCollection.find(onlyDocumentsReferencing(id), limitFive(session)).toArray();
-      //           const failures: string[] = referringDocuments.map(
-      //             (document) => `Resource ${document.resourceName} with identity '${JSON.stringify(document.documentIdentity)}'`,
-      //           );
-      // deleteResult = {
-      //   result: 'DELETE_FAILURE_REFERENCE',
-      //   failureMessage: `Delete failed due to existing references to document: ${failures.join(',')}`,
-      // };
-      //           await session.abortTransaction();
-      //           return;
-      //         }
+      // Check for any references to the document to be deleted
+      const referencesResult = await client.query(getCheckIsReferencedDocumentSql(id));
+
+      // Abort on validation failure
+      if (referencesResult.rows[0].exists) {
+        Logger.debug(
+          `postgres.repository.Delete.deleteDocumentById: Deleting document id ${id} failed due to existing references`,
+          traceId,
+        );
+
+        // Get the DocumentIdentities of up to five referring documents for failure message purposes
+        const referringDocuments = await client.query(getReferencedByDocumentSql(id));
+        const failures: string[] = referringDocuments.rows.map(
+          (document) => `Resource ${document.resource_name} with identity '${JSON.stringify(document.document_identity)}'`,
```

Comment:
We may need to be more subtle than this - in a future ticket though.

In other words, in what situations would it be unwise to reveal the document identity for documents that referred to the one being deleted?

1. Referring document is owned by a different client.
2. Referring document has PII in the identity.

On the second point, that might not be an issue if we can resolve the first point. After all, if the client "owns" the referring document, then why should we be concerned about an HTTP response containing identity information? In theory they send the body with that PII in the first place!